content/references/rippled-api/api-conventions/response-formatting.md

```diff
@@ -87,3 +89,84 @@ HTTP Status: 200 OK
 ```
 
 <!-- MULTICODE_BLOCK_END -->
+
+
+## API Warnings
+
+When the response contains a `warnings` array, each member of the array represents a separate warning from the server. Each such **Warning Object** contains the following fields:
+
+| `Field`   | Type   | Description                                             |
+|:----------|:-------|:--------------------------------------------------------|
+| `id`      | Number | A unique numeric code for this warning message.         |
+| `message` | String | A human-readable string describing the cause of this message. Do not write software that relies the contents of this message; use the `id` (and `details`, if applicable) to identify the warning instead. |
+| `details` | Object | _(May be omitted)_ Additional information about this warning. The contents vary depending on the type of warning. |
+
+The following reference describes all possible warnings.
+
+### 1001. Unsupported amendments have reached majority
+
+Example warning:
+
+```json
+"warnings" : [
+  {
+    "details" : {
+      "expected_date" : 864030,
+      "expected_date_UTC" : "2000-Jan-11 00:00:30.0000000 UTC"
+    },
+    "id" : 1001,
+    "message" : "One or more unsupported amendments have reached majority. Upgrade to the latest version before they are activated to avoid being amendment blocked."
+  }
+]
+```
+
+This warning indicates that the one or more [amendments](amendments.html) to the XRP Ledger protocol are scheduled to become enabled if they maintain support from trusted validators, but the current server does not have an implementation for those amendments. If those amendments become enabled, the current server will become [amendment blocked](amendments.html#amendment-blocked), so you should [upgrade to the latest `rippled` version](install-rippled.html) as soon as possible.
+
+The server only sends this warning if the client is [connected as an admin](get-started-with-the-rippled-api.html#admin-access).
+
+This warning includes a `details` field with the following fields:
+
+| Field               | Value  | Description                                   |
+|:--------------------|:-------|:----------------------------------------------|
+| `expected_date`     | Number | The time that the first unsupported amendment is expected to become enabled, in [seconds since the Ripple Epoch][]. |
```

Comment:
(Being pedantic here) - When is Number used and when is Integer used? Is it possible to have a non-integer number here?

Reply:
I'm phasing out the use of "Integer" for purely JSON data structures, as JSON doesn't make a distinction between number types; it has only one number type.